server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -917,7 +917,8 @@ protected boolean docValuesByDefault() {
     @Override
     protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
         super.doXContentBody(builder, includeDefaults, params);
-        if (includeDefaults || (mappedFieldType.isSearchable() && fieldType.indexOptions() != Defaults.FIELD_TYPE.indexOptions())) {
+        if ((includeDefaults && fieldType.indexOptions() != IndexOptions.NONE)
+            || (mappedFieldType.isSearchable() && fieldType.indexOptions() != Defaults.FIELD_TYPE.indexOptions())) {
```

Comment:
I struggle to parse this. why do we need to look at both indexOptions and isSearchable? When iis the second condition true in reality?

Reply:
Hm, you're right, we don't need the `isSearchable` bit there - will update, thanks!

Reply:
Have pushed a change which simplifies this